### PR TITLE
feat(api): implement logging api

### LINKED ISSUE #84: Logging package: Connect to message broker

### LINKED ISSUE #90: Logging package: Unit tests

### PR DESCRIPTION
**Pull request by jharvey10**
This closes out the remainder of the v0 logging epic (along with #193)

Closes #84, Closes #90

### CODE REVIEW

File: packages/api/src/main/logging/Logging.ts [hunk 1]
```diff
@@ -4,34 +4,109 @@
...
+    await this.log(logEntry)
   }
 
   public async info(message: string) {
-    this.winstonLogger.info({ label: this.component, message })
+    const logEntry = this.createMessage(message, 'info')
+    await this.log(logEntry)
+  }
+
+  public async warn(message: string | Error) {
```
Comment (francinelucca) [inline packages/api/src/main/logging/Logging.ts:59]: are these intuitive enough or should we also add jsdocs?

File: packages/api/src/main/logging/Logging.ts [hunk 2]
```diff
@@ -4,34 +4,181 @@
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
-import { Logger } from 'winston'
+import chalk from 'chalk'
 
-import { getRunMode } from '../run-mode'
-import { getWinstonLogger } from './winston-utils'
+import { EventMessage, MessagingClient } from '../messaging'
+import { DEBUG } from '../messaging/constants' // TODO: fix this
```
Comment (francinelucca) [inline packages/api/src/main/logging/Logging.ts:10]: what is this referring to? can we do a more explicit comment?

File: packages/api/src/main/logging/Logging.ts [hunk 3]
```diff
@@ -4,34 +4,109 @@
...
+  private readonly isRemoteLoggingEnabled: boolean
+  private readonly messagingClient: MessagingClient
+  private readonly service: string
 
-  constructor(component: string, winstonLogger?: Logger) {
+  constructor(serviceName: string, component: string) {
     this.component = component
-    this.winstonLogger = winstonLogger || getWinstonLogger(getRunMode())
+    // TODO: this should be based on an envvar indicating whether the service is running in a test
+    // environment or a prod environment
```
[the remark this answers is not in the document]
Comment (jdharvey-ibm) [reply packages/api/src/main/logging/Logging.ts:47]: This is kind of hinting at the idea that we will need some sort of indicator of which environment we're running in, but I haven't really thought it through yet. I didn't want to do too much in this realm without talking with you and the rest of the team to see what needs we have (if any) in this area.

Probably next sprint we should try to resolve as many of the todos in the repo as we can.

File: packages/api/src/main/logging/Logging.ts (hunk 2 above)
Comment (jdharvey-ibm) [reply packages/api/src/main/logging/Logging.ts:10]: haha, this is another reminder to myself of a var that should be pulled from the common infrastructure for envvars that you've started building out.

**Review by francinelucca — DISMISSED** [review]

**Review by mattrosno — DISMISSED** [review]

**Review by francinelucca — APPROVED** [review]

**Pull request merged**